utt/include/coin.hpp

```diff
@@ -0,0 +1,181 @@
+// UTT
+//
+// Copyright (c) 2020-2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the sub-component's license, as noted in the LICENSE
+// file.
+
+#pragma once
+#include "globalParams.hpp"
+#include "client.hpp"
+#include "commitment.hpp"
+#include "types.hpp"
+#include <memory>
+#include <optional>
+
+namespace libutt {
+class Coin;
+}
+namespace libutt::api {
+class Client;
+namespace operations {
+class Burn;
+class Mint;
+class Transaction;
+class Budget;
+}  // namespace operations
+
+class Coin {
+  /**
+   * @brief Represent a UTT coin. A coin can be either a normal value coin or a budget coin. A budget coin is a coin
+   * that limits the amount of annonymus transferred tokens.
+   *
+   */
+ public:
+  enum Type { Normal = 0x0, Budget };
```

Comment:
Is 'Normal' the best term for such a coin?

Reply:
This is the term that was used until now, so I would prefer not to change it